misk/src/main/kotlin/misk/web/metadata/guice/GuiceMetadata.kt

```diff
@@ -0,0 +1,163 @@
+package misk.web.metadata.guice
+
+import com.google.inject.Binding
+import com.google.inject.Injector
+import com.google.inject.Key
+import jakarta.inject.Inject
+import misk.web.metadata.Metadata
+import misk.web.metadata.MetadataProvider
+import misk.web.metadata.toFormattedJson
+import wisp.moshi.adapter
+import wisp.moshi.defaultKotlinMoshi
+
+internal data class GuiceMetadata(
+  val guice: GuiceMetadataProvider.Metadata
+) : Metadata(
+  metadata = guice,
+  prettyPrint = "Total Bindings: ${guice.all.size}\n\n" + defaultKotlinMoshi
+    .adapter<Map<String, Map<String, String>>>()
+    .toFormattedJson(guice.grouped),
+  descriptionString = "Direct injection bindings, powered by Guice. This metadata is work-in-progress."
+)
+
+internal class GuiceMetadataProvider : MetadataProvider<GuiceMetadata> {
+  @Inject lateinit var injector: Injector
+
+  override val id = "guice"
+
+  internal data class Metadata(
```

Comment:
I wonder if this data can ever get too big to a point where we might need to page through?

Reply:
Yeah might be a risk though I did strip a lot of data out of the `prettyPrint` version so that should likely be fine.

I agree though that the Metadata primitive could end up requiring paging, will need to give some thought on how best to do that.